qiskit_nature/second_q/operators/sparse_label_op.py

```diff
@@ -0,0 +1,140 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Sparse Label Operator base class."""
+
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import Mapping, Iterator
+from numbers import Number
+import cmath
+import numpy as np
+
+from qiskit.quantum_info.operators.mixins import LinearMixin, AdjointMixin, TolerancesMixin
+
+
+class SparseLabelOp(LinearMixin, AdjointMixin, TolerancesMixin, ABC):
+    """The Sparse Label Operator base class."""
+
+    def __init__(self, data: Mapping[str, complex], register_length: int, **kwargs):
+        """
+        Args:
+            data: Data for operator comprising string key to coeff value mapping
+            register_length: Length of register needed for data
+        """
+        self._data: Mapping[str, complex] = {}
+        if kwargs.get("_dont_copy", False):
```

Comment:
What is the purpose of hiding this inside `kwargs`? Can we not just add this explicitly?
We can enforce it to be used as a keyword argument by inserting `*` behind the `register_length` argument. I will make a suggestion to show how this is done